stdlib/public/Reflection/TypeRefBuilder.cpp

```diff
@@ -196,6 +196,11 @@ bool TypeRefBuilder::getFieldTypeRefs(
                      - FD.second->ReflectionString.SectionOffset;
     auto FieldName = Field.getFieldName(FieldOffset);
 
+    // Make sure that the field name falls within the boundaries of
+    // the reflectionstring section.
+    if (FieldOffset >= FD.second->ReflectionString.Metadata.size())
+      continue;
+
```

Comment:
@mikeash 

It looks like the API doesn't account for the possibility of an error at all, which is I guess because it was built with the assumption that it would always be called from within a program, where the metadata is guaranteed to be correct by the compiler.

Should we challenge that assumption for clients such as LLDB, and add error handling to the API?